packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -509,28 +511,51 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // only if it is first failure, otherwise fallback to get call.
         if (usePost) {
             try {
-                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, true);
+                const odspSnapshot = await this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, true);
+                return odspSnapshot;
             } catch (error) {
                 const errorType = error.errorType;
                 if ((errorType === DriverErrorType.authorizationError || errorType === DriverErrorType.incorrectServerResponse) && tokenFetchOptions.refresh === false) {
                     throw error;
                 }
                 this.logger.sendErrorEvent({ eventName: "TreeLatest_FallBackToGetRequest" }, error);
-                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+                return this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, false);
             }
         } else {
-            return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+            return this.fetchSnapshotWithTimeout(snapshotOptions, tokenFetchOptions, false);
+        }
+    }
+
+    private async fetchSnapshotWithTimeout(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions, usePost: boolean) {
+        assert(snapshotOptions.timeout !== undefined, "Timeout should be provided for setting a limit");
+        const abortController = new AbortController();
+        setTimeout(
+            () => {
+                abortController.abort();
+            },
+            snapshotOptions.timeout,
+        );
+        try {
+            const odspSnapshot = await this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, usePost, abortController);
+            return odspSnapshot;
+        } catch (error) {
+            if (error.errorType === OdspErrorType.snapshotTooBig || error.errorType === OdspErrorType.fetchTimeout) {
+                const snapshotOptionsWithoutBlobs: ISnapshotOptions = { ...snapshotOptions, blobs: undefined };
```

Comment:
It would be great to clear timeout here.
But more important - clear snapshot size restriction:
1. if it came from driver, not from host - clear it, we should try to load skeleton, even if it's bigger what driver thinks is reasonable (as otherwise we fail at all). Alternatively, we can proceed with such restrictions, but we need to ensure that on failure we can fall back to load file by reading tree by tree (how r11s workflow does it) 
2. If restriction came from host (Teams), we have to fail right here, not to proceed. Restriction is due to worries about bandwidth impact, not perf, so we have to respect that choice, not try circumvent that choice.

Reply:
Please see my comment earlier about user settings - we need to respect user settings (blobs, mds, etc.).
It's fine to put lower limits to fallback to other method of loading (no blobs), because driver believes not enough limitation is in place. But we neither can fail end-to-end process if driver put smaller limits above (rmd), neither we can circumvent total bandwidth restriction